Derkachenko.Anna/lab2/host/host.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+#include "client.hpp"
+#include "conn.hpp"
+#include "message.hpp"
+#include "utils.hpp"
+#include "wolf.hpp"
+#include <iostream>
+#include <semaphore.h>
+#include <signal.h>
+#include <stdexcept>
+#include <sys/mman.h>
+#include <unistd.h>
+
+class Host
+{
+public:
+    Host()
+    {
+        semHostMsg = (sem_t*)mmap(NULL, sizeof(sem_t), PROT_READ | PROT_WRITE, MAP_ANON | MAP_SHARED, -1, 0);
```

Comment:
зачем создавать в общей памяти области под семафоры? почему нельзя просто создать семафоры и передать указатели на него клиенту?

Reply:
Исходя из документации по sem_init (https://man7.org/linux/man-pages/man3/sem_init.3.html).
Цитирую:
"If pshared is nonzero, then the semaphore is shared between processes, and should be located in a region of shared memory (see shm_open(3), mmap(2), and shmget(2)).  (Since a child created by fork(2) inherits its parent's memory mappings, it can also access the semaphore.)  Any process that can access the shared memory region can operate on the semaphore using sem_post(3), sem_wait(3), and so on."

Reply:
окей, принято, спасибо за детальные пояснения